pkg/eval/builtin_fn_env.go

```diff
@@ -34,7 +34,7 @@ var ErrNonExistentEnvVar = errors.New("non-existent environment variable")
 // Unset an environment variable. Example:
 //
 // ```elvish-transcript
-// ~> set E:X = foo
+// ~> set-env X foo
```

Comment:
My preference is to keep the original statement, with a comment, and add the `set-env` statement. I think it's important to recognize that the `E:` namespace exists in these examples. Which obviously applies to closely related examples such as `del E:PATH` versus `unset-env PATH`. The `*-env` commands exist to support indirect env var names; e.g., `var y = X; set-env $y foo`.

Reply:
The other `*-env` command docs don't reference the `E:` namespace. Do you suggest something like this:

```
~> set E:X = foo # could also use e.g. set-env X foo
```